p2p/chain.py

```diff
@@ -589,19 +529,9 @@ def request_receipts(self, target_td: int, headers: List[BlockHeader]) -> int:
             header_request: Dict[str, Any]) -> None:
         self.logger.debug("Peer %s made header request: %s", peer, header_request)
 
-        try:
-            block_numbers = await self._get_block_numbers_for_request(
-                header_request['block_number_or_hash'], header_request['max_headers'],
-                header_request['skip'], header_request['reverse'])
-        except HeaderNotFound:
-            self.logger.debug(
-                "Peer %r requested starting header %r that is unavailable, returning nothing",
-                peer,
-                header_request['block_number_or_hash'],
-            )
-            block_numbers = []
-
-        headers = [header async for header in self._generate_available_headers(block_numbers)]
+        headers = await lookup_headers(
+            self.db, header_request['block_number_or_hash'], header_request['max_headers'],
```

Comment:
These call signatures are quite large.  What do you think about something like the following.

```
class PeerRequestHandler:
    def __init__(self, db, logger, cancel_token):
        self.db = db
        ...

    def lookup_headers(self, ...):
        return lookup_headers(db=self.db, ..., logger=self.logger, token=self.cancel_token)

    ...
```

Then the `BaseChainSyncer` classes can just set an instance of this class as a local variable and the call turns into `headers = await self.handler.lookup_headers(bn_or_hash, max_headers, skip, reverse)` and eliminates the extra 3 arguments that are always the same.

Reply:
Yeah, that's a nice improvement